src/seedu/addressbook/data/person/Name.java

```diff
@@ -67,7 +67,7 @@ public int hashCode() {
      */
     public boolean isSimilar(Name other) {
         
-        if (this.equals(other)){
+        if (this.fullName.toLowerCase().equals(((Name)other).fullName.toLowerCase())){
```

Comment:
You can take a look at `equalsIgnoreCase` method, to make this line more beautiful. :)